addon/modifiers/scroll-into-view.js

```diff
@@ -0,0 +1,22 @@
+import { modifier } from 'ember-modifier';
+
+export default modifier(function scrollIntoView(
+  element,
+  positional,
+  named = {}
+) {
+  const { options, shouldScroll } = named;
+  let hasBeenRemoved;
+
+  const trustedPromise = Promise.resolve(shouldScroll);
+
+  trustedPromise.then((result) => {
+    if (result && element && !hasBeenRemoved) {
```

Comment:
Would rename `result` to something semantic.

Reply:
If I change `trustedPromise` to `shouldScrollPromise` then I can change change `result` to `shouldScrollValue`. How does that sound? (Sorry, naming things is clearly not my forte)

Reply:
sounds better :)